cis/libs/connection.py

```diff
@@ -0,0 +1,52 @@
+"""Centralized handling of AWS Boto sessions for change integration."""
+
+import boto3
+
+class Connect(object):
+    def __init__(self, connection_type, service=None, region='us-west-2', profile='default'):
+        """
+        :param connection_type: session|client|resource
+        :param service: kms|dynamodb|other
+        :param region: us-east-1|us-west-2
+        :param profile: default but supports others
+        """
+        self.service = service
+        self.region = region
+        self.connection_type = connection_type
+        self.profile = profile
+
+        try:
+            # Attempt to init default boto session and raise if fails
+            boto3.setup_default_session(profile_name=self.profile)
+        except Exception as e:
```

Comment:
This was intentional to support a desktop use case.  But... I've eliminated the block now so it will just blow up if no boto profile exists.